scene/main/viewport.cpp

```diff
@@ -3807,6 +3811,25 @@ float Viewport::get_texture_mipmap_bias() const {
 
 #endif // _3D_DISABLED
 
+void Viewport::_propagate_world_2d_changed(Node *p_node) {
+	if (p_node != this) {
+		if (Object::cast_to<CanvasItem>(p_node)) {
+			p_node->notification(CanvasItem::NOTIFICATION_WORLD_2D_CHANGED);
+		} else {
+			Viewport *v = Object::cast_to<Viewport>(p_node);
+			if (v) {
+				if (v->world_2d.is_valid()) {
```

Comment:
```suggestion
			if (v && v->world_2d.is_valid()) {
```
Can be simplified.

Reply:
I guess this was written like this to be consistent with 3D:
```cpp
Viewport *v = Object::cast_to<Viewport>(p_node);
if (v) {
        if (v->world_3d.is_valid() || v->own_world_3d.is_valid()) {
                return;
        }
}
```
Which would need to become `if (v && (v->world_3d.is_valid() || v->own_world_3d.is_valid()))` too.